lib/RealWlanInterface.cpp

```diff
@@ -384,6 +384,7 @@ void RealWlanInterface::OnScanComplete()
         if (m_scanPromise)
         {
             m_scanPromise->set_value({std::move(results), ScanStatus::Completed});
+            m_scanPromise = std::nullopt;
```

Comment:
This will fulfill the bound promise synchronously, correct?

Reply:
No, this is totally asynchronous. Destroying the promise will cause it to release its reference to the shared state. However, the shared state won't be destroyed as long as there is a future referencing it.

Reply:
Yes, what I meant is that once `set_value()` has completed (synchronous bit), the future (I mistakenly "bound promise" 😒) will be fulfilled.